File: sort/selectionSort/eg2.java
import java.util.*;
class SelectionSort<T>
{
public  void sort(T [] arr, Object c)
{
int n=arr.length;
for(int e=0;e<(n-1);e++)
{
int si=e;
for(int f=e+1;f<n;f++)
{
if(c.compare((T)arr[f],(T)arr[e])<0)
{
si=f;
}
}
T g=arr[e];
arr[e]=arr[si];
arr[si]=g;
}
}
}
class test
{
public static void main(String gg[])
{
Integer [] arr=new Integer[10];
for(int i=0;i<10;i++)
{
Scanner sc=new Scanner(System.in);
System.out.print("Enter a number :");
arr[i]=sc.nextInt();
}
SelectionSort<Integer> ss;
Integer t=new Integer(1);
//Comparator<Integer> r=(Comparator<Integer>)t;
ss.sort(arr,t);
for(int i=0;i<10;i++) System.out.println(arr[i]);
}
}

File: sort/bubbleSort/eg2.cpp
#include<iostream>
using namespace std;
int main()
{
int req;
cout<<"Enter requirement :";
cin>>req;
int *arr=new int[req];
for(int i=0;i<req;i++)
{
cout<<"Enter a number :";
cin>>arr[i];
}
for(int m=req-1;m>=0;m--)
{
for(int e=0,f=1;e<m;e++,f++)
{
if(arr[f]<arr[e])
{
int g=arr[e];
arr[e]=arr[f];
arr[f]=g;
}
}
}
for(int i=0;i<req;i++) cout<<arr[i]<<endl;
return 0;
}